src/sentry/snuba/metrics/extraction.py

```diff
@@ -80,12 +81,15 @@ class OnDemandMetricSpecVersioning:
 
     spec_versions = [
         SpecVersion(1),
+        SpecVersion(2, {"include_environment_tag"}),
     ]
 
     @classmethod
     def get_query_spec_version(cls: Any, organization_id: int) -> SpecVersion:
         """Return spec version based on feature flag enabled for an organization."""
-        _ = Organization.objects.get_from_cache(id=organization_id)
+        org = Organization.objects.get_from_cache(id=organization_id)
+        if features.has("organizations:on-demand-metrics-query-spec-version-two", org):
+            return cls.spec_versions[1]
         return cls.spec_versions[0]
 
```

Comment:
Not blocking, but doing this by array lookup is a little strange since that feature flag should _only_ be for SpecVersion(2), I think this can be hardcoded, which will make cleanup more obvious as well.

Reply:
Yeah. I used to have a different approach that did not require any changes.